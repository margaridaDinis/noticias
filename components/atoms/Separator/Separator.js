import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import Colors from '../../../constants/Colors';

const Separator = ({ color = Colors.borderColor, marginTop = 0, marginBottom = 0 }) => (
  <View style={{ height: 1, backgroundColor: color, marginTop, marginBottom }} />
);

Separator.propTypes = {
  color: PropTypes.string,
  marginTop: PropTypes.number,
  marginBottom: PropTypes.number,
};

export default Separator;
