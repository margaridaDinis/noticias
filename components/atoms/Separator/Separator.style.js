import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  separator: {
    height: 1,
  },
});

export default styles;
