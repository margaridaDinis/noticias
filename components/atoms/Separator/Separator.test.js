import React from 'react';
import { render } from '@testing-library/react-native';
import Separator from './Separator';

describe('<Separator />', () => {
  it('renders without props', () => {
    const { queryByTestId } = render(<Separator />);

    expect(queryByTestId('separator').props.style).toMatchSnapshot();
  });

  it('renders with custom backgroundColor', () => {
    const { queryByTestId } = render(<Separator color="gold" />);

    expect(queryByTestId('separator').props.style).toMatchSnapshot();
  });

  it('renders with custom margins', () => {
    const { queryByTestId } = render(
      <Separator marginBottom={16} marginTop={8} />,
    );

    expect(queryByTestId('separator').props.style).toMatchSnapshot();
  });
});
