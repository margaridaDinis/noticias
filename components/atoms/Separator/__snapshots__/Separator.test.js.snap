// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Separator /> renders with custom backgroundColor 1`] = `
Array [
  Object {
    "height": 1,
  },
  Object {
    "backgroundColor": "gold",
    "marginBottom": 0,
    "marginTop": 0,
  },
]
`;

exports[`<Separator /> renders with custom margins 1`] = `
Array [
  Object {
    "height": 1,
  },
  Object {
    "backgroundColor": "#e0e0e0",
    "marginBottom": 16,
    "marginTop": 8,
  },
]
`;

exports[`<Separator /> renders without props 1`] = `
Array [
  Object {
    "height": 1,
  },
  Object {
    "backgroundColor": "#e0e0e0",
    "marginBottom": 0,
    "marginTop": 0,
  },
]
`;
